test/unit/krane/kubectl_test.rb

```diff
@@ -348,6 +348,16 @@ def test_retry_delay_backoff
     end
   end
 
+  def test_debug_level_output_log_uses_correct_encoding
+    logger.level = ::Logger::DEBUG
+    good = "hélas"
+    bad = good.dup.force_encoding(Encoding::US_ASCII)
```

Comment:
This obviously isn't how the encoding goes bad in the wild, but this test does reproduce the stack trace from the bug report. Note that we have to `dup` because string literals are frozen, and `force_encoding` is considered a modification even though it doesn't touch the characters themselves.